docs/attributes.md

```diff
@@ -0,0 +1,210 @@
+# Attributes
+
+As pointed out in
+[Attributes as Patterns](<../design-decisions/2018-07-25 Attributes as patterns.md>),
+attributes are encoded as Kore patterns that only use application and string
+literals.
+
+## Axiom Pattern Attributes
+
+Kore.Step.AxiomPatterns
+
+Attributes specific to interpreting axiom patterns.
+
+### Assoc
+
+Associativity axiom
+
+```
+assoc{}()
+```
+
+Example, List concatenation
+
+```
+axiom{R} \equals{SortList{}, R} (
+  Lbl'Unds'List'Unds'{}(
+    Lbl'Unds'List'Unds'{}(K1:SortList{},K2:SortList{}),
+    K3:SortList{}),
+  Lbl'Unds'List'Unds'{}(
+    K1:SortList{},
+    Lbl'Unds'List'Unds'{}(K2:SortList{},K3:SortList{}))) [assoc{}()]
+```
+
+### Comm
+
+Commutativity axiom
+
+```
+comm{}()
+```
+
+Example: Map concatenation
+
+```
+axiom{R} \equals{SortMap{}, R} (
+  Lbl'Unds'Map'Unds'{}(K1:SortMap{},K2:SortMap{}),
+  Lbl'Unds'Map'Unds'{}(K2:SortMap{},K1:SortMap{})) [comm{}()]
+```
+
+
+### HeatCool
+
+Heating and cooling axiom
+
+```
+heat{}()
+cool{}()
+```
+
+In each execution step, the execution strategy first heats the configuration,
+applies the normal rules (without `heat{}()` and `cool{}()`), and cools the
+result.
+
+
+### Idem
+
+Idempotency axiom
+
+```
+idem{}()
+```
+
+Example: concatenation of identical Sets
+
+```
+axiom{R} \equals{SortSet{}, R} (
+  Lbl'Unds'Set'Unds'{}(K:SortSet{},K:SortSet{}),K:SortSet{}) [idem{}()]
+```
+
+### ProductionID
+
+```
+productionID{}("id")
+```
+
+### Trusted
+
+```
+trusted{}()
+```
+
+Spec axioms with this attribute are excluded from verification.
+
+### Unit
+
+Unit element axiom
+
+```
+unit{}()
+```
+
+
+## Stepper Attributes
+
+Kore.Step.StepperAttributes
+
+Attributes used during Kore execution.
+
+### Constructor
+
+```
+constructor{}()
+```
+
+A symbol is injective if it is given the `injective` attribute, the
+`constructor` attribute, or the `sortInjection` attribute.
+
+### Function
+
+```
+function{}()
+```
+
+A symbol is a function if it is given the `function` attribute or if it is
+functional.
+
+### Functional
+
+```
+functional{}()
+```
+
+A symbol is functional if it is given the `functional` attribute or the
+`sortInjection` attribute.
+
+### Hook
+
+```
+hook{}("BUILTIN.name")
+```
+
+The string `"BUILTIN.name"` is used to lookup for known builtins.
```

Comment:
I'd add a reference to https://github.com/kframework/kore/blob/master/docs/hooks.md